### PR TITLE
Update search form

### PR DESCRIPTION
**Pull request by icefleen**

### CODE REVIEW

File: src/stores/SearchStore.ts [hunk 1]
```diff
@@ -45,7 +45,7 @@ export type SearchResult = EventTreeNode | EventMessage;
 
 export type SearchHistory = {
 	timestamp: number;
-	results: Array<SearchResult>;
+	results: Record<string, Array<SearchResult>>;
```
Comment (rzvdv) [inline src/stores/SearchStore.ts:48]: Some users already have search results saved in localstorage and their will not be able to use app unless they clear local storage themselves

i got
Uncaught TypeError: Cannot read property 'messageId' of undefined SearchPanelResults.tsx:64

File: src/components/search-panel/SearchPanelForm.tsx [hunk 2]
```diff
@@ -134,20 +132,53 @@ const SearchPanelForm = (props: Props) => {
...
+						<div className='search-panel__fields'>
+							<div className='filter-row'>
+								<div className='filter-row__label'>Search for</div>
+								<SearchTypeSwitcher formType={formType} setFormType={setFormType} />
+							</div>
+							<div className='filter-row'>
+								<div className='filter-row__label'>Time Limit</div>
+								<FilterDatetimeInput {...endTimestampInput} />
+							</div>
+							{config.map(rowConfig => (
```
Comment (rzvdv) [inline src/components/search-panel/SearchPanelForm.tsx:161]: I thought we move result count limit somewhere. currently it takes whole row, but it doesnt need it. I dont know maybe move it to same row as `Search for` ? @YourWeebyNeighbor what do you think?

File: src/components/search-panel/SearchPanelResults.tsx [hunk 3]
```diff
@@ -49,16 +51,31 @@ const SearchPanelResults = (props: SearchPanelResultsProps) => {
...
 	function computeKey(index: number) {
-		const item = resultGroups[index][0] || 'key';
+		const [, results] = resultGroups[index];
+		const item = results[0];
 
 		return isEventNode(item) ? item.eventId : item.messageId;
 	}
 
+	const onScroll = (e: React.UIEvent<HTMLDivElement, UIEvent>) => {
+		if (e.target === e.currentTarget) {
```
Comment (rzvdv) [inline src/components/search-panel/SearchPanelResults.tsx:66]: It's bugged when after collapsing form, results take more space and items take less space then results height, so after onScrollEvent occurs its scrollTop is 0 and its collapsing

File: src/stores/SearchStore.ts (hunk 1 above)
Comment (icefleen) [reply src/stores/SearchStore.ts:54]: We'll think about it. @YourWeebyNeighbor says we can validate localStorage data and clear it if we got an invalid data format. It'll be resolved in the future

File: src/components/search-panel/SearchPanelForm.tsx (hunk 2 above)
Comment (icefleen) [reply src/components/search-panel/SearchPanelForm.tsx]: Done
image

File: src/components/search-panel/SearchPanelResults.tsx (hunk 3 above)
Comment (icefleen) [reply src/components/search-panel/SearchPanelResults.tsx]: I removed this functionality temporarily

**Review by YourWeebyNeighbor — APPROVED** [review]

**Pull request merged**